[2025-09-14T09:00:15.557+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_etl_pipeline.gold_transform scheduled__2025-09-14T07:00:00+00:00 [queued]>
[2025-09-14T09:00:15.566+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_etl_pipeline.gold_transform scheduled__2025-09-14T07:00:00+00:00 [queued]>
[2025-09-14T09:00:15.566+0000] {taskinstance.py:1361} INFO - Starting attempt 6 of 6
[2025-09-14T09:00:15.578+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): gold_transform> on 2025-09-14 07:00:00+00:00
[2025-09-14T09:00:15.584+0000] {standard_task_runner.py:57} INFO - Started process 7775 to run task
[2025-09-14T09:00:15.589+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spotify_etl_pipeline', 'gold_transform', 'scheduled__2025-09-14T07:00:00+00:00', '--job-id', '622', '--raw', '--subdir', 'DAGS_FOLDER/spotify_dag.py', '--cfg-path', '/tmp/tmpir_x1qyj']
[2025-09-14T09:00:15.592+0000] {standard_task_runner.py:85} INFO - Job 622: Subtask gold_transform
[2025-09-14T09:00:15.638+0000] {task_command.py:415} INFO - Running <TaskInstance: spotify_etl_pipeline.gold_transform scheduled__2025-09-14T07:00:00+00:00 [running]> on host 0d0cd054d690
[2025-09-14T09:00:15.997+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_etl_pipeline' AIRFLOW_CTX_TASK_ID='gold_transform' AIRFLOW_CTX_EXECUTION_DATE='2025-09-14T07:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-09-14T07:00:00+00:00'
[2025-09-14T09:00:16.004+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/etl/db_utils.py", line 11, in run_sql_scripts
    cur.execute(sql)
psycopg2.errors.DependentObjectsStillExist: cannot drop table gold.fact_listening_history because other objects depend on it
DETAIL:  view gold.view_top_songs depends on table gold.fact_listening_history
view gold.view_total_listening_by_day depends on table gold.fact_listening_history
view gold.view_total_listening_overall depends on table gold.fact_listening_history
view gold.view_top_artists depends on table gold.fact_listening_history
view gold.view_monthly_genre_distribution depends on table gold.fact_listening_history
view gold.view_monthly_genre_top_artists depends on table gold.fact_listening_history
HINT:  Use DROP ... CASCADE to drop the dependent objects too.

[2025-09-14T09:00:16.013+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spotify_etl_pipeline, task_id=gold_transform, execution_date=20250914T070000, start_date=20250914T090015, end_date=20250914T090016
[2025-09-14T09:00:16.035+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 622 for task gold_transform (cannot drop table gold.fact_listening_history because other objects depend on it
DETAIL:  view gold.view_top_songs depends on table gold.fact_listening_history
view gold.view_total_listening_by_day depends on table gold.fact_listening_history
view gold.view_total_listening_overall depends on table gold.fact_listening_history
view gold.view_top_artists depends on table gold.fact_listening_history
view gold.view_monthly_genre_distribution depends on table gold.fact_listening_history
view gold.view_monthly_genre_top_artists depends on table gold.fact_listening_history
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
; 7775)
[2025-09-14T09:00:16.059+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-09-14T09:00:16.078+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
